### ISSUE TITLE
Add reference-based InteractionModel.SlotType models

### ISSUE THREAD

**Issue by dennis-kuypers-gcx** [stoiveyp/Alexa.NET.Management#72]
SlotType - instead of containing value literals - may contain a reference to a catalog when using [reference-based catalogs](https://developer.amazon.com/en-US/docs/alexa/smapi/reference-based-catalog-management.html).

Sample from the interaction model `interactionModel.languageModel.types[]`
```
{
  "name": "<name>",
  "valueSupplier": {
    "type": "CatalogValueSupplier",
    "valueCatalog": {
      "catalogId": "amzn1.ask.interactionModel.catalog.<catalog-id>",
      "version": "<version>"
    }
  }
}
```

This could be a problem when updating models via `Get-Update`
```
var model = await _management.InteractionModel.Get(...);
// <changes to model>
await _management.InteractionModel.Update(..., model);
```

**Comment by stoiveyp** [comment]
Slot types do support a CatalogValueSupplier

The ValueSupplier object is part of the slot type version. ValueSupplier is built to be extensible, but the code for the CatalogValueSupplier is here:
https://github.com/stoiveyp/Alexa.NET.Management/blob/master/Alexa.NET.Management/SlotType/CatalogValueSupplier.cs

The test for this is here:
https://github.com/stoiveyp/Alexa.NET.Management/blob/master/Tests/Alexa.NET.Management.Tests/SlotTypeTests.cs#L127

Hope that helps :+1:

**Comment by dennis-kuypers-gcx** [comment]
The interaction model slot type class is `Alexa.NET.Management.InteractionModel.SlotType`. This is currently not deserializing correctly when we get a payload from my question, right? (I haven't found a converter for it)

**Comment by stoiveyp** [comment]
Ahh thank you - I see what you mean now!

This has now been fixed in v5.4.1
https://www.nuget.org/packages/Alexa.NET.Management/5.4.1
